interop-tests/src/lib.rs

```diff
@@ -0,0 +1,319 @@
+use chrono::offset::{TimeZone, Utc};
+use data_encoding::HEXLOWER;
+use serde_derive::Deserialize;
+use std::collections::{BTreeMap, HashMap};
+use std::fs::{self, File};
+use std::path::{Path, PathBuf};
+use tuf::crypto::{Ed25519PrivateKey, HashAlgorithm, KeyType, PrivateKey, SignatureScheme};
+use tuf::interchange::JsonPretty;
+use tuf::metadata::{
+    MetadataPath, MetadataVersion, Role, RootMetadataBuilder, SnapshotMetadataBuilder, TargetPath,
+    TargetsMetadataBuilder, TimestampMetadataBuilder, VirtualTargetPath,
+};
+use tuf::repository::{FileSystemRepository, FileSystemRepositoryBuilder, RepositoryStorage};
+use walkdir::WalkDir;
+
+// These structs and functions are necessary to parse keys.json, which contains the keys
+// used by go-tuf to generate the equivalent metadata. We use the same keys to facilitate
+// compatibility testing.
+
+#[derive(Clone, Deserialize)]
+struct KeyValue {
+    #[serde(rename = "public")]
+    _public: String,
+    private: String,
+}
+
+#[derive(Clone, Deserialize)]
+struct TestKeyPair {
+    #[serde(rename = "keytype")]
+    _keytype: KeyType,
+    #[serde(rename = "scheme")]
+    _scheme: SignatureScheme,
+    #[serde(rename = "keyid_hash_algorithms")]
```

Comment:
Do we still need this for backwards compat?

Reply:
Nope, I removed it.